### ISSUE TITLE
Loss spikes / numerical instabilities when training CEBRA models on Apple Silicon (M2)

### ISSUE THREAD

**Issue by stes** [AdaptiveMotorControlLab/CEBRA#220]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Bug description

When training with `mps` backend on torch, we get spikes in the loss. This is probably an upstream issue in pytorch. I am attaching code to repro plus test outputs for training on cpu vs. mps.

### Operating System

Mac OS

### CEBRA version

cebra version

### Device type

mps (Apple Silicon M2)

### Steps To Reproduce

``` python
import matplotlib.pyplot as plt
import cebra.datasets
import cebra.integrations.sklearn.metrics as cebra_sklearn_metrics

data = cebra.datasets.init("rat-hippocampus-single-achilles")
model = cebra.CEBRA(batch_size = 1024, max_iterations = 1000, verbose = True, device="cpu")
model.fit(data.neural)

plt.plot(model.state_dict_["log"]["total"])
plt.xlabel("Steps")
plt.ylabel("InfoNCE loss")
plt.show()

model = cebra.CEBRA(batch_size = 1024, max_iterations = 1000, verbose = True, device="mps")
model.fit(data.neural)

plt.plot(model.state_dict_["log"]["total"])
plt.xlabel("Steps")
plt.ylabel("InfoNCE loss")
plt.show()
```

Which gives the following output (top is cpu, bottom is mps)

<img width="862" alt="Image" src="https://github.com/user-attachments/assets/5c0559c2-03de-417c-aaf8-50cd85f93fdc" />

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/daptiveMotorControlLab/CEBRA/blob/main/CODE_OF_CONDUCT.md)

**Comment by stes** [comment]
just checked again.

``` python
import matplotlib.pyplot as plt
import cebra.datasets
import cebra.integrations.sklearn.metrics as cebra_sklearn_metrics

import cebra, torch, numpy
for module in cebra, torch, numpy:
    print(module.__name__, module.__version__)

data = cebra.datasets.init("rat-hippocampus-single-achilles")
model = cebra.CEBRA(batch_size = 1024, max_iterations = 1000, verbose = True, device="cpu")
model.fit(data.neural)

plt.plot(model.state_dict_["log"]["total"])
plt.xlabel("Steps")
plt.ylabel("InfoNCE loss")
plt.savefig("cpu.png")
plt.close()

model = cebra.CEBRA(batch_size = 1024, max_iterations = 1000, verbose = True, device="mps")
model.fit(data.neural)

plt.plot(model.state_dict_["log"]["total"])
plt.xlabel("Steps")
plt.ylabel("InfoNCE loss")
plt.savefig("mps.png")
plt.close()
```

**cpu**

<img width="596" alt="Image" src="https://github.com/user-attachments/assets/1a5a04c1-10f8-4b71-82cb-4b93ec02b7c4" />

**mps**

<img width="580" alt="Image" src="https://github.com/user-attachments/assets/dd3a0e47-84d3-4fa6-80ec-2fbc67535b76" />

using

```
cebra 0.6.0a2
torch 2.7.0
numpy 1.26.4
```

seems to be fixed!

using `torch 2.6.0` instead:

<img width="589" alt="Image" src="https://github.com/user-attachments/assets/2be52c51-7007-4ae4-a3b1-533c3ba0e9fa" />

using `numpy 2.0.0` instead:

<img width="577" alt="Image" src="https://github.com/user-attachments/assets/388bb590-3274-4534-9aa4-5826292772d5" />
